docs/architecture/architecture.md

```diff
@@ -0,0 +1,130 @@
+
+#### Core
+
+Consists of the designs and logic pertaining to the transport, authentication, and ordering layers of the IBC protocol, the fundamental pieces. 
+
+##### Client (ICS 02)
+
+Clients encapsulate all of the verification methods of another IBC-enabled chain in order to ensure that the other chain adheres to the IBC protocol and does not exhibit misbehaviour. Clients "track" the metadata of the other chain's blocks, and each chain has a client for every other chain that it communicates with. 
+
+##### Connection (ICS 03)
+
+Connections associate a chain with another chain by connecting a client on the local chain with a client on the remote chain. This association is pair-wise unique and is established between two chains following a 4-step handshake process. 
+
+##### Channel (ICS 04)
+
+Channels are an abstraction layer that facilitate communication between applications and the chains those applications are built upon. One important function that channels can fulfill is guaranteeing that data packets sent between an application and its chain are well-ordered. 
+
+##### Port (ICS 05)
+
+The port standard specifies an allocation scheme by which modules can bind to uniquely-named ports allocated by the IBC handler in order to facilitate module-to-module traffic. These ports are used to open channels and can be transferred or released by the module which originally bound them.
+
+##### Commitment (ICS 23)
+
+Commitments (sometimes called _vector commitments_) define an efficient cryptographic construction to prove inclusion or non-inclusion of values in at particular paths in state. This scheme provides a guarantee of a particular state transition that has occurred on one chain which can be verified on another chain.
+
+#### Applications
+
+Consists of various packet encoding and processing semantics which underpin the various types of transactions that users can perform on any IBC-compliant chain.
+
+##### Fungible Token Transfer (ICS 20)
+
+Specifies the packet data structure, state machine handling logic, and encoding details used for transferring fungible tokens between IBC chains. This process preserves asset fungibility and ownership while limiting the impact of Byzantine faults. 
+
+#### Clients
+
+Consists of implementations of client verification algorithms (following the base client interface that is defined in `Core`) for specific types of chains. A chain uses these verification algorithms to verify the state of a remote chain.
+
+##### Tendermint (ICS 07)
+
+The Tendermint client implements a client verification algorithm for blockchains which use the Tendermint consensus algorithm. This enables state machines of various sorts replicated using the Tendermint consensus algorithm to interface with other replicated state machines or solo machines over IBC. 
+
+#### Relayer
```

Comment:
having Relayer and Relayer (ICS 18) is a bit confusing

Reply:
I couldn't really think of a clean way of addressing this. It's a bit awkward because the directory is named `relayer` and it contains a relayer implementation; the naming follows the specification, so it can't really be changed. I opted to simply have the ICS designation first and then the name. I don't know if this really addresses your concern here, but hopefully it helps a little.